inc/class-component.php

```diff
@@ -245,6 +258,32 @@ public function set_config_by_key( string $key, $value ): self {
 		return $this;
 	}
 
+	/**
+	 * Get the config schema.
+	 *
+	 * @return array
+	 */
+	public function get_config_schema() {
+		return $this->config_schema;
+	}
+
+	/**
+	 * Set schema describing config.
+	 *
+	 * @param array $config_schema Schema array.
+	 * @return self
+	 */
+	public function set_config_schema( array $config_schema ): self {
```

Comment:
We may want to make this private and only set schema as an internal implementation detail, but I think it's fine to leave as is for now.